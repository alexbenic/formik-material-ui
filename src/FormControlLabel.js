/* @flow */
import FormControlLabel from '@material-ui/core/FormControlLabel';

import createComponent from './createComponent';

export default createComponent(
  FormControlLabel,
  ({ field, form: { touched, errors, isSubmitting }, ...props }) => {
    const { name } = field;

    return {
      ...props,
      ...field,
      error: touched[name] && !!errors[name],
      disabled: isSubmitting,
    };
  }
);
