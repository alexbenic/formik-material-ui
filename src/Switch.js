/* @flow */
import Switch from '@material-ui/core/Switch';

import createComponent from './createComponent';

export default createComponent(
  Switch,
  ({ field, form: { isSubmitting }, ...props }) => ({
    disabled: isSubmitting,
    ...props,
    ...field,
    // TODO: is this the correct way?
    value: field.value ? 'checked' : '',
  })
);
