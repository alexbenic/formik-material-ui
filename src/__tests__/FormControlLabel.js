/* @flow */
import React from 'react';
import { Formik, Field, Form } from 'formik';
import renderer from 'react-test-renderer';
import Switch from '@material-ui/core/Switch';

import FormControlLabel from '../FormControlLabel';

test('FormControlLabel Renders Correctly', () => {
  const component = renderer.create(
    <Formik onSubmit={() => null}>
      <Form>
        <Field
          name="test"
          label="Text"
          component={FormControlLabel}
          control={<Switch />}
        />
      </Form>
    </Formik>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
