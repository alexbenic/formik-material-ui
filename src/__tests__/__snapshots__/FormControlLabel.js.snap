// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormControlLabel Renders Correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    className="MuiFormControlLabel-root-1"
    error={undefined}
    onBlur={[Function]}
  >
    <span
      className="MuiSwitch-root-4"
    >
      <span
        className="MuiButtonBase-root-23 MuiIconButton-root-17 MuiSwitchBase-root-13 MuiSwitch-switchBase-7 MuiSwitch-colorSecondary-10"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={undefined}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label-22"
        >
          <span
            className="MuiSwitch-icon-5"
          />
          <input
            checked={false}
            className="MuiSwitchBase-input-16"
            disabled={false}
            id={undefined}
            name="test"
            onChange={[Function]}
            tabIndex={undefined}
            type="checkbox"
            value={undefined}
          />
        </span>
        <span
          className="MuiTouchRipple-root-26"
        />
      </span>
      <span
        className="MuiSwitch-bar-12"
      />
    </span>
    <span
      className="MuiTypography-root-33 MuiTypography-body1-42 MuiFormControlLabel-label-3"
    >
      Text
    </span>
  </label>
</form>
`;
